client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/RemoteSchemasAdminImpl.java

```diff
@@ -0,0 +1,386 @@
+    public CompletionStage<SchemaOpResult> createOrUpdateAsync(Schema schema, boolean force) {
+        if (force) {
+            // Update regardless of the actual value
+            return protostreamCache.putAsync(schema.getName(), schema.getContent())
+                    .thenApply(r -> r == null ? SchemaOpResult.CREATED : SchemaOpResult.UPDATED)
+                    .exceptionally(ex -> {
+                        if (ex != null) {
+                            throw new InfinispanException(ex.getMessage(), ex);
+                        }
+                        return SchemaOpResult.NONE;
+                    });
+        }
+
+        return CompletionStages.handleAndCompose(protostreamCache.getWithMetadataAsync(schema.getName()), (actualSchema, ex) -> {
+            if (ex != null) {
+                throw new InfinispanException(ex.getMessage(), ex);
+            }
+
+            if (actualSchema == null) {
+                // Might not create if other process creates meanwhile
+                return createAsync(schema);
+            }
+
+            // Update. might not update anything if other process updates it before
+            return updateAsync(schema);
+        });
+    }
+
+
+    @Override
+    public Optional<Schema> get(String schemaName) {
+        MetadataValue<String> withMetadata = protostreamCache.getWithMetadata(schemaName);
+        return getSchemaContent(schemaName, withMetadata);
+    }
+
+    @Override
+    public CompletionStage<Optional<Schema>> getAsync(String schemaName) {
+        return protostreamCache.getWithMetadataAsync(schemaName)
+                .thenApply(e -> getSchemaContent(schemaName, e))
+                .exceptionally(ex -> {
+                    if (ex != null) {
+                        throw new InfinispanException(ex.getMessage(), ex);
+                    }
+                    return Optional.empty();
+                });
+    }
+
+    private static Optional<Schema> getSchemaContent(String schemaName, MetadataValue<String> withMetadata) {
+        if (withMetadata != null && withMetadata.getValue() != null) {
+            return Optional.of(Schema.buildFromStringContent(schemaName, withMetadata.getValue()));
+        }
+
+        return Optional.empty();
+    }
+
+    @Override
+    public SchemaOpResult remove(String name) {
+        return remove(name, false);
+    }
+
+    @Override
+    public SchemaOpResult remove(String name, boolean force) {
+        if (force) {
+            String existing = protostreamCache.withFlags(FORCE_RETURN_VALUE).remove(name);
+            if (existing == null) {
+                log.infof(String.format("Schema %s not found", name));
+            }
+            return existing != null ? SchemaOpResult.DELETED : SchemaOpResult.NONE;
+        }
+
+        Map<String, byte[]> params = Map.of("schema", string(name));
+        String opResult = await(operationDispatcher.execute(operationsFactory.executeOperation("@@schemas@delete", params)), timeout);
```

Comment:
I think timeout is used internally to cancel the request. I don't think you need to set it here.